### PR TITLE
Release 1.6.0

### PR DESCRIPTION
**Pull request by tcojean**
Finalize the release 1.6.0 in the develop branch.

Also, bump to 1.7.0 for develop.

### CODE REVIEW

**Comment by tcojean** [comment]
General question: should I try to summarize the release notes as we did earlier by grouping them by topic? Or do we stay with the current format, assuming we have shorter release cycles? This release is shorter than before so going through the changelog is not so bad in the current state.

**Comment by upsj** [comment]
I would probably try to reorder the changelog a bit and combine items that describe iterations on the same functionality, but other than that, I think it's already a pretty good list

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,111 @@ git log --first-parent
...
+  + Intel compiler: 2018+
+  + Apple LLVM: 8.0+
+  + NVHPC: 22.7+
+  + Cray Compiler: 14.0.1+
+  + CUDA module: CUDA 9.2+ or NVHPC 22.7+
+  + HIP module: ROCm 4.5+
+  + DPC++ module: Intel OneAPI 2021.3 with oneMKL and oneDPL. Set the CXX compiler to `dpcpp`.
++ Windows
+  + MinGW and Cygwin: GCC 5.5+
+  + Microsoft Visual Studio: VS 2019
```
Comment (yhmtsai) [inline CHANGELOG.md:44]:
```suggestion
  + MinGW: GCC 5.5+
  + Microsoft Visual Studio: VS 2019+
```
I think we can also remove cygwin

File: CHANGELOG.md [hunk 2]
```diff
@@ -11,6 +11,111 @@ git log --first-parent
...
++ Linux and macOS
+  + GCC: 5.5+
+  + clang: 3.9+
+  + Intel compiler: 2018+
+  + Apple LLVM: 8.0+
+  + NVHPC: 22.7+
+  + Cray Compiler: 14.0.1+
+  + CUDA module: CUDA 9.2+ or NVHPC 22.7+
+  + HIP module: ROCm 4.5+
+  + DPC++ module: Intel OneAPI 2021.3 with oneMKL and oneDPL. Set the CXX compiler to `dpcpp`.
```
Comment (yhmtsai) [inline CHANGELOG.md:41]:
```suggestion
  + DPC++ module: Intel OneAPI 2021.3+ with oneMKL and oneDPL. Set the CXX compiler to `dpcpp`.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -11,6 +11,111 @@ git log --first-parent
...
++ ROCm 4.0+ -> 4.5+ after [#1303](https://github.com/ginkgo-project/ginkgo/pull/1303)
+
+### Interface Changes
++ Due to internal changes, `ConcreteExecutor::run` will now always throw if the corresponding module for the `ConcreteExecutor` is not build [#1234](https://github.com/ginkgo-project/ginkgo/pull/1234)
++ Changes the constructor of `experimental::distributed::Vector` to only accept local vectors as `std::unique_ptr` [#1284](https://github.com/ginkgo-project/ginkgo/pull/1284)
++ Changes some default parameter for the `solver::MultiGrid`. In particular, the smoother default to on iteration of `Ir` with `Jacobi` preconditioner, and the coarse grid solver uses the new direct solver with LU factorization. [#1291](https://github.com/ginkgo-project/ginkgo/pull/1291) [#1327](https://github.com/ginkgo-project/ginkgo/pull/1327)
++ Adds a more expresive callback to the `iteration_complete` event and deprecates the old ones. [#1288](https://github.com/ginkgo-project/ginkgo/pull/1288) [#1327](https://github.com/ginkgo-project/ginkgo/pull/1327)
+
+### Deprecations
+
```
Comment (yhmtsai) [inline CHANGELOG.md:58]:
```suggestion
```

**Review by MarcelKoch — APPROVED** [review]
I personally think the changelog is fine as it is. The highlight section at the top is a good summary.

File: CHANGELOG.md [hunk 4]
```diff
@@ -11,6 +11,111 @@ git log --first-parent
...
++ Deprecated less expressive `iteration_complete` event. Users are advised to now implement the function `void iteration_complete(const LinOp* solver, const LinOp* b, const LinOp* x, const size_type& it, const LinOp* r, const LinOp* tau, const LinOp* implicit_tau_sq, const array<stopping_status>* status, bool stopped)` [#1288](https://github.com/ginkgo-project/ginkgo/pull/1288)
+
+### Added Features
++ Add a distributed Schwarz preconditioner. [#1248](https://github.com/ginkgo-project/ginkgo/pull/1248)
```
Comment (yhmtsai) [inline CHANGELOG.md:62]:
```suggestion
+ Added a distributed Schwarz preconditioner. [#1248](https://github.com/ginkgo-project/ginkgo/pull/1248)
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -11,6 +11,111 @@ git log --first-parent
...
++ Flexible Gmres solver [#1244](https://github.com/ginkgo-project/ginkgo/pull/1244)
++ Enabled Gmres solver for distributed matrices and vectors [#1201](https://github.com/ginkgo-project/ginkgo/pull/1201)
++ Example that uses Kokkos to assemble the matrix [#1216](https://github.com/ginkgo-project/ginkgo/pull/1216)
++ Symbolic LU factorization allowing the `gko::experimental::factorization::Lu` and `gko::experimental::solver::Direct` classes to be used for matrices with non-symmetric sparsity pattern [#1210](https://github.com/ginkgo-project/ginkgo/pull/1210)
++ Numerical Cholesky factorization [#1215](https://github.com/ginkgo-project/ginkgo/pull/1215)
++ Add GCR solver [#1239](https://github.com/ginkgo-project/ginkgo/pull/1239)
```
Comment (yhmtsai) [inline CHANGELOG.md:68]:
```suggestion
+ Added GCR solver [#1239](https://github.com/ginkgo-project/ginkgo/pull/1239)
```
because the others uses `added`. Or, change the others to `add`

File: CHANGELOG.md [hunk 6]
```diff
@@ -11,6 +11,111 @@ git log --first-parent
...
++ Annotate solver iterations in `ProfilerHook` [#1290](https://github.com/ginkgo-project/ginkgo/pull/1290)
++ Allow passing smart pointers in place of raw pointers to most matrix functions. This means that things like `vec->compute_norm2(x.get())` or `vec->compute_norm2(lend(x))` can be simplified to `vec->compute_norm2(x)` [#1279](https://github.com/ginkgo-project/ginkgo/pull/1279) [#1261](https://github.com/ginkgo-project/ginkgo/pull/1261)
++ Catch overflows in prefix sum operations, which makes Ginkgo's operations much less likely to crash. This also improves the performance of the prefix sum kernel [#1303](https://github.com/ginkgo-project/ginkgo/pull/1303)
++ Make the installed GinkgoConfig.cmake file relocatable and follow more best practices [#1325](https://github.com/ginkgo-project/ginkgo/pull/1325)
++ Allow using the profiler hooks and inline input strings in benchmarks [#1342](https://github.com/ginkgo-project/ginkgo/pull/1342)
+
+### Fixes
++ Fix the wrong type and pass real-number value with device_type to devices [#1253](https://github.com/ginkgo-project/ginkgo/pull/1253)
```
Comment (upsj) [inline CHANGELOG.md:94]: This only seems relevant to Ginkgo developers, maybe we should remove it?

File: CHANGELOG.md [hunk 7]
```diff
@@ -11,6 +11,111 @@ git log --first-parent
...
++ Fix OpenMPI version check [#1200](https://github.com/ginkgo-project/ginkgo/pull/1200)
++ Fix hwloc related slowdown in MPI applications [#1230](https://github.com/ginkgo-project/ginkgo/pull/1230)
```
Comment (upsj) [inline CHANGELOG.md:96]: I think we can be more explicit: 
```suggestion
+ Disable thread pinning with GPU executors due to poor performance [#1230](https://github.com/ginkgo-project/ginkgo/pull/1230)
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -11,6 +11,111 @@ git log --first-parent
...
++ Fix hwloc version detection [#1266](https://github.com/ginkgo-project/ginkgo/pull/1266)
++ Fix PAPI detection is non-implicit include directories [#1268](https://github.com/ginkgo-project/ginkgo/pull/1268)
```
Comment (upsj) [inline CHANGELOG.md:98]:
```suggestion
+ Fix PAPI detection in non-implicit include directories [#1268](https://github.com/ginkgo-project/ginkgo/pull/1268)
```

File: CHANGELOG.md [hunk 9]
```diff
@@ -11,6 +11,111 @@ git log --first-parent
...
++ Fix PAPI support for newer PAPI versions: [#1321](https://github.com/ginkgo-project/ginkgo/pull/1321)
++ Fix pkg-config file generation for library paths outside prefix [#1271](https://github.com/ginkgo-project/ginkgo/pull/1271)
++ Fix build failures with ROCm 5.4 due to missing Thrust includes [#1214](https://github.com/ginkgo-project/ginkgo/pull/1214)
++ Fix build failures with CUDA 12 due to deprecated functionality [#1235](https://github.com/ginkgo-project/ginkgo/pull/1235)
++ Fix build failures with OneAPI 6 due to changed namespace structure [#1251](https://github.com/ginkgo-project/ginkgo/pull/1251)
```
Comment (upsj) [inline CHANGELOG.md:103]: I think we could group those together

File: CHANGELOG.md [hunk 10]
```diff
@@ -11,6 +11,111 @@ git log --first-parent
...
++ Fix the mpi cxx type binding by c binding [#1306](https://github.com/ginkgo-project/ginkgo/pull/1306)
++ Fix the additional residual check in IR [#1307](https://github.com/ginkgo-project/ginkgo/pull/1307)
++ Fix the cuSPARSE CSR SpMM issue on single strided vector when SM >= 70 and cuda >= 11.6 [#1322](https://github.com/ginkgo-project/ginkgo/pull/1322)
++ Fix the stencil matrix creation for benchmarking [#1305](https://github.com/ginkgo-project/ginkgo/pull/1305)
++ Fix Isai generation for large sparsity powers [#1327](https://github.com/ginkgo-project/ginkgo/pull/1327)
++ Fix Ginkgo compilation and test with NVHPC >= 22.7 [#1331](https://github.com/ginkgo-project/ginkgo/pull/1331)
+
+### CI-related
++ Remove our own gitlab status update [#1231](https://github.com/ginkgo-project/ginkgo/pull/1231)
++ Add wiki check [#1218](https://github.com/ginkgo-project/ginkgo/pull/1218)
```
Comment (upsj) [inline CHANGELOG.md:117]: those two don't seem relevant to users?

File: CHANGELOG.md [hunk 11]
```diff
@@ -11,6 +11,111 @@ git log --first-parent
...
++ Enables using generated stencil matrices (5-pt, 7-pt, 9-pt, and 27-pt) for benchmarks [#1204](https://github.com/ginkgo-project/ginkgo/pull/1204)
++ Added distributed benchmarks (multi-vector blas, spmv, solver) [#1204](https://github.com/ginkgo-project/ginkgo/pull/1204)
++ Added benchmarks for CSR sorting and lookup [#1219](https://github.com/ginkgo-project/ginkgo/pull/1219)
++ Added timer for MPI benchmarks that reports the longest time [#1217](https://github.com/ginkgo-project/ginkgo/pull/1217)
++ Added support for non-default streams in CUDA and HIP executors [#1236](https://github.com/ginkgo-project/ginkgo/pull/1236)
++ Added different timer statistical method for output [#1294](https://github.com/ginkgo-project/ginkgo/pull/1294)
++ Added METIS integration for nested dissection reordering [#1296](https://github.com/ginkgo-project/ginkgo/pull/1296)
++ Added SuiteSparse AMD integration for fillin-reducing reordering [#1328](https://github.com/ginkgo-project/ginkgo/pull/1328)
++ Added Csr mixed spmv support [#1319](https://github.com/ginkgo-project/ginkgo/pull/1319)
```
Comment (upsj) [inline CHANGELOG.md:82]:
```suggestion
+ Added Csr mixed-precision spmv support [#1319](https://github.com/ginkgo-project/ginkgo/pull/1319)
```

File: CHANGELOG.md [hunk 12]
```diff
@@ -11,6 +11,111 @@ git log --first-parent
...
++ It is now possible to enable host buffers for MPI communications at runtime even if the compile option `GINKGO_FORCE_GPU_AWARE_MPI` is set. [#1228](https://github.com/ginkgo-project/ginkgo/pull/1228)
++ Enables using generated stencil matrices (5-pt, 7-pt, 9-pt, and 27-pt) for benchmarks [#1204](https://github.com/ginkgo-project/ginkgo/pull/1204)
++ Added distributed benchmarks (multi-vector blas, spmv, solver) [#1204](https://github.com/ginkgo-project/ginkgo/pull/1204)
++ Added benchmarks for CSR sorting and lookup [#1219](https://github.com/ginkgo-project/ginkgo/pull/1219)
++ Added timer for MPI benchmarks that reports the longest time [#1217](https://github.com/ginkgo-project/ginkgo/pull/1217)
++ Added support for non-default streams in CUDA and HIP executors [#1236](https://github.com/ginkgo-project/ginkgo/pull/1236)
++ Added different timer statistical method for output [#1294](https://github.com/ginkgo-project/ginkgo/pull/1294)
```
Comment (upsj) [inline CHANGELOG.md:79]:
```suggestion
+ Added ´-timer_method=min|max|average|median` for benchmark timing summary [#1294](https://github.com/ginkgo-project/ginkgo/pull/1294)
```

File: CHANGELOG.md [hunk 13]
```diff
@@ -11,6 +11,111 @@ git log --first-parent
...
++ Fix build failures with CUDA 12 due to deprecated functionality [#1235](https://github.com/ginkgo-project/ginkgo/pull/1235)
++ Fix build failures with OneAPI 6 due to changed namespace structure [#1251](https://github.com/ginkgo-project/ginkgo/pull/1251)
++ Fix runtime failures for one-sided MPI wrapper functions observed on some OpenMPI versions [#1249](https://github.com/ginkgo-project/ginkgo/pull/1249)
++ Fix incorrect read for skew-symmetric MatrixMarket files with explicit diagonal entries [#1272](https://github.com/ginkgo-project/ginkgo/pull/1272)
++ Fix handling of missing diagonal entries in symbolic factorizations [#1263](https://github.com/ginkgo-project/ginkgo/pull/1263)
++ Fix segmentation fault in benchmark matrix construction [#1299](https://github.com/ginkgo-project/ginkgo/pull/1299)
++ Fix the mpi cxx type binding by c binding [#1306](https://github.com/ginkgo-project/ginkgo/pull/1306)
++ Fix the additional residual check in IR [#1307](https://github.com/ginkgo-project/ginkgo/pull/1307)
++ Fix the cuSPARSE CSR SpMM issue on single strided vector when SM >= 70 and cuda >= 11.6 [#1322](https://github.com/ginkgo-project/ginkgo/pull/1322)
```
Comment (yhmtsai) [inline CHANGELOG.md:110]:
```suggestion
+ Fix the cuSPARSE CSR SpMM issue on single strided vector when cuda >= 11.6 [#1322](https://github.com/ginkgo-project/ginkgo/pull/1322) [#1331](https://github.com/ginkgo-project/ginkgo/pull/1331)
```
SM < 70 is failed at cuda >= 11.7 though

File: CHANGELOG.md [hunk 14]
```diff
@@ -11,6 +11,111 @@ git log --first-parent
...
++ Symbolic factorization benchmark [#1302](https://github.com/ginkgo-project/ginkgo/pull/1302)
++ The `ProfilerHook` logger allows annotating the Ginkgo execution (apply, operations, ...) for profiling frameworks like NVTX, ROCTX and TAU. [#1055](https://github.com/ginkgo-project/ginkgo/pull/1055)
++ `ProfilerHook::created_(nested_)summary` allows the generation of a lightweight runtime profile over all Ginkgo functions written to a user-defined stream [#1270](https://github.com/ginkgo-project/ginkgo/pull/1270) for both host and device timing functionality [#1313](https://github.com/ginkgo-project/ginkgo/pull/1313)
++ It is now possible to enable host buffers for MPI communications at runtime even if the compile option `GINKGO_FORCE_GPU_AWARE_MPI` is set. [#1228](https://github.com/ginkgo-project/ginkgo/pull/1228)
++ Enables using generated stencil matrices (5-pt, 7-pt, 9-pt, and 27-pt) for benchmarks [#1204](https://github.com/ginkgo-project/ginkgo/pull/1204)
```
Comment (upsj) [inline CHANGELOG.md:74]:
```suggestion
+ Added generator for stencil matrices (5-pt, 7-pt, 9-pt, and 27-pt) to benchmarks [#1204](https://github.com/ginkgo-project/ginkgo/pull/1204)
```

File: CHANGELOG.md [hunk 15]
```diff
@@ -11,6 +11,111 @@ git log --first-parent
 
 Please visit our wiki [Changelog](https://github.com/ginkgo-project/ginkgo/wiki/Changelog) for unreleased changes.
 
+## Version 1.6.0
+
+The Ginkgo team is proud to announce the new Ginkgo minor release 1.6.0. This release brings new features such as:
+- Several building blocks for GPU-resident sparse direct solvers like the
+  symbolic Cholesky factorization, numerical LU factorization, ...,
```
Comment (upsj) [inline CHANGELOG.md:18]:
```suggestion
- Several building blocks for GPU-resident sparse direct solvers like
  symbolic and numerical LU and Cholesky factorization, ...,
```

**Review by yhmtsai — APPROVED** [review]
Did you also check the pull request before wiki check because we do not introduce the wiki check right after the release 1.5?

**Review by upsj — COMMENTED** [review]
I feel like the changelog currently reads like it was written by a lot of different people (it was, after all 😄 ), but maybe we can smoothen things out a bit by using the same sentence structure for each section. E.g. we don't need to say `added ...` in the section `Added Features`

**Comment by tcojean** [comment]
I did some small changes. I should have integrated all comments.

File: CHANGELOG.md [hunk 16]
```diff
@@ -11,6 +11,107 @@ git log --first-parent
...
++ Adds a more expresive callback to the `iteration_complete` event and deprecates the old ones. [#1288](https://github.com/ginkgo-project/ginkgo/pull/1288) [#1327](https://github.com/ginkgo-project/ginkgo/pull/1327)
+
+### Deprecations
++ Deprecated less expressive `iteration_complete` event. Users are advised to now implement the function `void iteration_complete(const LinOp* solver, const LinOp* b, const LinOp* x, const size_type& it, const LinOp* r, const LinOp* tau, const LinOp* implicit_tau_sq, const array<stopping_status>* status, bool stopped)` [#1288](https://github.com/ginkgo-project/ginkgo/pull/1288)
+
+### Added Features
++ Added a distributed Schwarz preconditioner. [#1248](https://github.com/ginkgo-project/ginkgo/pull/1248)
++ Add GCR solver [#1239](https://github.com/ginkgo-project/ginkgo/pull/1239)
++ Flexible Gmres solver [#1244](https://github.com/ginkgo-project/ginkgo/pull/1244)
++ Enabled Gmres solver for distributed matrices and vectors [#1201](https://github.com/ginkgo-project/ginkgo/pull/1201)
```
Comment (pratikvn) [inline CHANGELOG.md:68]:
```suggestion
+ Enable Gmres solver for distributed matrices and vectors [#1201](https://github.com/ginkgo-project/ginkgo/pull/1201)
```

File: CHANGELOG.md [hunk 17]
```diff
@@ -11,6 +11,107 @@ git log --first-parent
...
++ Example that uses Kokkos to assemble the matrix [#1216](https://github.com/ginkgo-project/ginkgo/pull/1216)
```
Comment (pratikvn) [inline CHANGELOG.md:69]:
```suggestion
+ An example that uses Kokkos to assemble the system matrix [#1216](https://github.com/ginkgo-project/ginkgo/pull/1216)
```

File: CHANGELOG.md [hunk 18]
```diff
@@ -11,6 +11,107 @@ git log --first-parent
...
++ Symbolic LU factorization allowing the `gko::experimental::factorization::Lu` and `gko::experimental::solver::Direct` classes to be used for matrices with non-symmetric sparsity pattern [#1210](https://github.com/ginkgo-project/ginkgo/pull/1210)
```
Comment (pratikvn) [inline CHANGELOG.md:70]:
```suggestion
+ A symbolic LU factorization allowing the `gko::experimental::factorization::Lu` and `gko::experimental::solver::Direct` classes to be used for matrices with non-symmetric sparsity pattern [#1210](https://github.com/ginkgo-project/ginkgo/pull/1210)
```

File: CHANGELOG.md [hunk 19]
```diff
@@ -11,6 +11,107 @@ git log --first-parent
...
++ Numerical Cholesky factorization [#1215](https://github.com/ginkgo-project/ginkgo/pull/1215)
```
Comment (pratikvn) [inline CHANGELOG.md:71]:
```suggestion
+ A numerical Cholesky factorization [#1215](https://github.com/ginkgo-project/ginkgo/pull/1215)
```

**Review by pratikvn — APPROVED** [review]
Some nits, otherwise looks good,

File: CHANGELOG.md [hunk 20]
```diff
@@ -11,6 +11,107 @@ git log --first-parent
...
+If you face an issue, please first check our [known issues page](https://github.com/ginkgo-project/ginkgo/wiki/Known-Issues) and the [open issues list](https://github.com/ginkgo-project/ginkgo/issues) and if you do not find a solution, feel free to [open a new issue](https://github.com/ginkgo-project/ginkgo/issues/new/choose) or ask a question using the [github discussions](https://github.com/ginkgo-project/ginkgo/discussions).
+
+Supported systems and requirements:
++ For all platforms, CMake 3.13+
++ C++14 compliant compiler
++ Linux and macOS
+  + GCC: 5.5+
+  + clang: 3.9+
+  + Intel compiler: 2018+
+  + Apple LLVM: 8.0+
```
[the remark this answers is not in the document]
Comment (tcojean) [reply CHANGELOG.md:39]: No idea. Only mac users can answer this. We probably should check it manually or bump it.

[the remark this answers is not in the document]
Comment (pratikvn) [reply CHANGELOG.md:39]: Yeah, I think we should just bump it. It is a pain to verify it as I think Apple forces you to install XCode for even specific Apple Clang versions, which is upwards of 10GB. In case someone is able to compile and run it on their machine for lower versions, then that is fine.

File: CHANGELOG.md [hunk 21]
```diff
@@ -11,6 +11,107 @@ git log --first-parent
...
+  + CUDA module: CUDA 9.2+, Microsoft Visual Studio
+  + OpenMP module: MinGW.
+
+### Version Support Changes
++ ROCm 4.0+ -> 4.5+ after [#1303](https://github.com/ginkgo-project/ginkgo/pull/1303)
++ Removed Cygwin pipeline and support [#1283](https://github.com/ginkgo-project/ginkgo/pull/1283)
+
+### Interface Changes
++ Due to internal changes, `ConcreteExecutor::run` will now always throw if the corresponding module for the `ConcreteExecutor` is not build [#1234](https://github.com/ginkgo-project/ginkgo/pull/1234)
++ Changes the constructor of `experimental::distributed::Vector` to only accept local vectors as `std::unique_ptr` [#1284](https://github.com/ginkgo-project/ginkgo/pull/1284)
```
Comment (upsj) [inline CHANGELOG.md:57]:
```suggestion
+ The constructor of `experimental::distributed::Vector` was changed to only accept local vectors as `std::unique_ptr` [#1284](https://github.com/ginkgo-project/ginkgo/pull/1284)
```

File: CHANGELOG.md [hunk 22]
```diff
@@ -11,6 +11,107 @@ git log --first-parent
...
++ Changes some default parameter for the `solver::MultiGrid`. In particular, the smoother defaults to one iteration of `Ir` with `Jacobi` preconditioner, and the coarse grid solver uses the new direct solver with LU factorization. [#1291](https://github.com/ginkgo-project/ginkgo/pull/1291) [#1327](https://github.com/ginkgo-project/ginkgo/pull/1327)
```
Comment (upsj) [inline CHANGELOG.md:58]:
```suggestion
+ The default parameters for the `solver::MultiGrid` were improved. In particular, the smoother defaults to one iteration of `Ir` with `Jacobi` preconditioner, and the coarse grid solver uses the new direct solver with LU factorization. [#1291](https://github.com/ginkgo-project/ginkgo/pull/1291) [#1327](https://github.com/ginkgo-project/ginkgo/pull/1327)
```

File: CHANGELOG.md [hunk 23]
```diff
@@ -11,6 +11,107 @@ git log --first-parent
...
++ Adds a more expresive callback to the `iteration_complete` event and deprecates the old ones. [#1288](https://github.com/ginkgo-project/ginkgo/pull/1288) [#1327](https://github.com/ginkgo-project/ginkgo/pull/1327)
```
Comment (upsj) [inline CHANGELOG.md:59]:
```suggestion
+ The `iteration_complete` event gained a more expressive overload with additional parameters, the old overloads were deprecated. [#1288](https://github.com/ginkgo-project/ginkgo/pull/1288) [#1327](https://github.com/ginkgo-project/ginkgo/pull/1327)
```

File: CHANGELOG.md [hunk 24]
```diff
@@ -11,6 +11,107 @@ git log --first-parent
...
++ Removed Cygwin pipeline and support [#1283](https://github.com/ginkgo-project/ginkgo/pull/1283)
+
+### Interface Changes
++ Due to internal changes, `ConcreteExecutor::run` will now always throw if the corresponding module for the `ConcreteExecutor` is not build [#1234](https://github.com/ginkgo-project/ginkgo/pull/1234)
++ The constructor of `experimental::distributed::Vector` was changed to only accept local vectors as `std::unique_ptr` [#1284](https://github.com/ginkgo-project/ginkgo/pull/1284)
++ The default parameters for the `solver::MultiGrid` were improved. In particular, the smoother defaults to one iteration of `Ir` with `Jacobi` preconditioner, and the coarse grid solver uses the new direct solver with LU factorization. [#1291](https://github.com/ginkgo-project/ginkgo/pull/1291) [#1327](https://github.com/ginkgo-project/ginkgo/pull/1327)
++ The `iteration_complete` event gained a more expressive overload with additional parameters, the old overloads were deprecated. [#1288](https://github.com/ginkgo-project/ginkgo/pull/1288) [#1327](https://github.com/ginkgo-project/ginkgo/pull/1327)
```
Comment (yhmtsai) [inline CHANGELOG.md:59]: only 2 and 4 are interface changes but the others are behavior changes?

File: CHANGELOG.md (hunk 13 above)
Comment (upsj) [reply CHANGELOG.md]: @tcojean This one still needs to be addressed

File: CHANGELOG.md [hunk 25]
```diff
@@ -11,6 +11,107 @@ git log --first-parent
...
++ A stencil matrices generator (5-pt, 7-pt, 9-pt, and 27-pt) for benchmarks [#1204](https://github.com/ginkgo-project/ginkgo/pull/1204)
++ Distributed benchmarks (multi-vector blas, spmv, solver) [#1204](https://github.com/ginkgo-project/ginkgo/pull/1204)
++ Benchmarks for CSR sorting and lookup [#1219](https://github.com/ginkgo-project/ginkgo/pull/1219)
++ A timer for MPI benchmarks that reports the longest time [#1217](https://github.com/ginkgo-project/ginkgo/pull/1217)
++ A `timer_method=min|max|average|median` for benchmark timing summary [#1294](https://github.com/ginkgo-project/ginkgo/pull/1294)
++ Support for non-default streams in CUDA and HIP executors [#1236](https://github.com/ginkgo-project/ginkgo/pull/1236)
++ METIS integration for nested dissection reordering [#1296](https://github.com/ginkgo-project/ginkgo/pull/1296)
++ SuiteSparse AMD integration for fillin-reducing reordering [#1328](https://github.com/ginkgo-project/ginkgo/pull/1328)
++ Csr mixed-precision spmv support [#1319](https://github.com/ginkgo-project/ginkgo/pull/1319)
```
Comment (upsj) [inline CHANGELOG.md:85]:
```suggestion
+ Csr mixed-precision SpMV support [#1319](https://github.com/ginkgo-project/ginkgo/pull/1319)
```

File: CHANGELOG.md [hunk 26]
```diff
@@ -11,6 +11,107 @@ git log --first-parent
...
++ `ProfilerHook::created_(nested_)summary` allows the generation of a lightweight runtime profile over all Ginkgo functions written to a user-defined stream [#1270](https://github.com/ginkgo-project/ginkgo/pull/1270) for both host and device timing functionality [#1313](https://github.com/ginkgo-project/ginkgo/pull/1313)
++ It is now possible to enable host buffers for MPI communications at runtime even if the compile option `GINKGO_FORCE_GPU_AWARE_MPI` is set. [#1228](https://github.com/ginkgo-project/ginkgo/pull/1228)
++ A stencil matrices generator (5-pt, 7-pt, 9-pt, and 27-pt) for benchmarks [#1204](https://github.com/ginkgo-project/ginkgo/pull/1204)
++ Distributed benchmarks (multi-vector blas, spmv, solver) [#1204](https://github.com/ginkgo-project/ginkgo/pull/1204)
++ Benchmarks for CSR sorting and lookup [#1219](https://github.com/ginkgo-project/ginkgo/pull/1219)
++ A timer for MPI benchmarks that reports the longest time [#1217](https://github.com/ginkgo-project/ginkgo/pull/1217)
++ A `timer_method=min|max|average|median` for benchmark timing summary [#1294](https://github.com/ginkgo-project/ginkgo/pull/1294)
```
Comment (upsj) [inline CHANGELOG.md:81]:
```suggestion
+ A `timer_method=min|max|average|median` flag for benchmark timing summary [#1294](https://github.com/ginkgo-project/ginkgo/pull/1294)
```

File: CHANGELOG.md [hunk 27]
```diff
@@ -11,6 +11,107 @@ git log --first-parent
 
 Please visit our wiki [Changelog](https://github.com/ginkgo-project/ginkgo/wiki/Changelog) for unreleased changes.
 
+## Version 1.6.0
+
+The Ginkgo team is proud to announce the new Ginkgo minor release 1.6.0. This release brings new features such as:
+- Several building blocks for GPU-resident sparse direct solvers like symbolic
+  and numerical LU and Cholesky factorization, symbolic Cholesky factorization,
+  numerical LU factorization, ...,
```
Comment (upsj) [inline CHANGELOG.md:19]:
```suggestion
  and numerical LU and Cholesky factorization, ...,
```

File: CHANGELOG.md [hunk 28]
```diff
@@ -11,6 +11,107 @@ git log --first-parent
...
++ A stencil matrices generator (5-pt, 7-pt, 9-pt, and 27-pt) for benchmarks [#1204](https://github.com/ginkgo-project/ginkgo/pull/1204)
++ Distributed benchmarks (multi-vector blas, spmv, solver) [#1204](https://github.com/ginkgo-project/ginkgo/pull/1204)
++ Benchmarks for CSR sorting and lookup [#1219](https://github.com/ginkgo-project/ginkgo/pull/1219)
++ A timer for MPI benchmarks that reports the longest time [#1217](https://github.com/ginkgo-project/ginkgo/pull/1217)
++ A `timer_method=min|max|average|median` for benchmark timing summary [#1294](https://github.com/ginkgo-project/ginkgo/pull/1294)
++ Support for non-default streams in CUDA and HIP executors [#1236](https://github.com/ginkgo-project/ginkgo/pull/1236)
++ METIS integration for nested dissection reordering [#1296](https://github.com/ginkgo-project/ginkgo/pull/1296)
++ SuiteSparse AMD integration for fillin-reducing reordering [#1328](https://github.com/ginkgo-project/ginkgo/pull/1328)
++ Csr mixed-precision spmv support [#1319](https://github.com/ginkgo-project/ginkgo/pull/1319)
++ A `with_loggers` function to all `Factory` parameters [#1337](https://github.com/ginkgo-project/ginkgo/pull/1337)
```
Comment (upsj) [inline CHANGELOG.md:86]:
```suggestion
+ A `with_loggers` function for all `Factory` parameters [#1337](https://github.com/ginkgo-project/ginkgo/pull/1337)
```

**Comment by upsj** [comment]
rebase!
